src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/AddAStandard/AddLocationController.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.Authorization.Mvc.Attributes;
+using SFA.DAS.Roatp.CourseManagement.Application.ProviderLocations.Queries.GetAllProviderLocations;
+using SFA.DAS.Roatp.CourseManagement.Domain.ApiModels;
+using SFA.DAS.Roatp.CourseManagement.Web.Infrastructure;
+using SFA.DAS.Roatp.CourseManagement.Web.Infrastructure.Authorization;
+using SFA.DAS.Roatp.CourseManagement.Web.Models;
+using SFA.DAS.Roatp.CourseManagement.Web.Models.AddAStandard;
+using SFA.DAS.Roatp.CourseManagement.Web.Services;
+
+namespace SFA.DAS.Roatp.CourseManagement.Web.Controllers.AddAStandard
+{
+    [DasAuthorize(new[] { "ProviderFeature.CourseManagement" }, Policy = nameof(PolicyNames.HasProviderAccount))]
+    public class AddLocationController : ControllerBase
+    {
+        public const string ViewPath = "~/Views/AddAStandard/AddCourseLocation.cshtml";
+        private readonly ILogger<AddLocationController> _logger;
+        private readonly IMediator _mediator;
+        private readonly ISessionService _sessionService;
+
+        public AddLocationController(IMediator mediator, ISessionService sessionService, ILogger<AddLocationController> logger)
+        {
+
+            _mediator = mediator;
+            _sessionService = sessionService;
+            _logger = logger;
+        }
+
+        [Route("{ukprn}/standards/{larsCode}/locations/add-new", Name = RouteNames.GetNewStandardAddProviderCourseLocation)]
+        [HttpGet]
+        public async Task<IActionResult> SelectAProviderlocation(int larsCode)
+        {
+            var model = await GetModel(larsCode);
+            if (model == null) return RedirectToRouteWithUkprn(RouteNames.GetAddStandardSelectStandard);
+            return View(ViewPath, model);
+        }
+
+        [Route("{ukprn}/standards/{larscode}/locations/add-new", Name = RouteNames.PostNewStandardAddProviderCourseLocation)]
```

Comment:
larscode has  not been used in route anywhere else. I would rename the route to `{ukprn}/standards/add/locations/add`

Reply:
done